src/js/protocols/webstm32.js

```diff
@@ -100,10 +100,31 @@ class STM32Protocol {
         serial.removeEventListener('disconnect', (event) => this.handleDisconnect(event.detail));
 
         if (disconnectionResult && this.rebootMode) {
-            DFU.requestPermission()
-            .then((device) => {
-                DFU.connect(device.path, this.hex, this.serialOptions);
-            });
+
+            // If the firmware_flasher does not start flashing, we need to ask for permission to flash
+            setTimeout(() => {
+                if (this.rebootMode) {
+                    console.log('STM32 Requesting permission for device');
+
+                    DFU.requestPermission()
+                    .then((device) => {
+
+                        if (device != null) {
+                            console.log('DFU request permission granted', device);
+                        } else {
+                            console.error('DFU request permission denied');
+                            this.rebootMode = 0;
+                            GUI.connect_lock = false;
+                        }
+                    })
+                    .catch (e => {
+                        console.error('DFU request permission failed', e);
+                        this.rebootMode = 0;
+                        GUI.connect_lock = false;
+                    });
+                }
+            }, 3000);
```

Comment:
Chris needed 5 seconds here (in NWjs version)

Reply:
Interesting. Here is almost immediate. I let enough time but not too much for the user that is waiting...

Reply:
Guess it was without flash on connect. Not sure how much it is needed anymore as I have never needed or used it before. Afaik it was used on F3 which we no longer support. We don't track the use of it so we just don't know :)